src/plugins/OEmbed/Components.tsx

```diff
@@ -71,38 +78,42 @@ export const OEmbed = (
   if ("html" in props) {
     const processedHtml = preprocessHtml(props.html);
     return (
-      <Article
-        data-src={props.attributes.src}
-        data-spoilers={props.attributes.spoilers}
-        data-width={props.attributes.width}
-        data-height={props.attributes.height}
-        data-source={getWebsiteNameFromUrl(props.meta.canonical)}
-        data-loaded={props.loaded}
-        dangerouslySetInnerHTML={{ __html: processedHtml }}
-        ref={onAttachNode}
-      />
+      <BlockBaseComponent {...props} className="embed" enclosingTag={"article"}>
+        <div
+          className={wrapperClass}
+          dangerouslySetInnerHTML={{ __html: processedHtml }}
+          data-source={getWebsiteNameFromUrl(props.meta.canonical)}
+          data-loaded={props.loaded}
+          ref={onAttachNode}
+        />
+      </BlockBaseComponent>
     );
   }
   return <>Unimplemented</>;
 };
 
-export const OEmbedPlaceholder = (props: OEmbedData) => {
+// Not sure if we actually want the BlockBaseComponent here,
```

Comment:
Until the size is settled again on the client, an embed should indeed take up the space that was previously saved. Not sure I understand correctly though.

Reply:
So this placeholder is only used in `renderHTML`, which as it doesn't create a NodeView will not actually load the iframe at any point, so using the BlockBaseComponent here will create a sized element that will not necessarily be filled unless the consumer implements fetching the embed themselves and injecting it.

I guess the real question is what do we want the output of `renderHTML` to be/be useful for? Presumably we are not going to use it on Boba directly, since we render content within a read-only editor. So would we want it to potentially be useful for data portability/sharing across platforms? If so, it should contain the data-attributes with the information on width and height we have gathered, but probably not the style setting width and height. The consumer should totally be in charge of how they render it, no?

Reply:
Yes, the idea would be that in e.g. a RSS feed you'd be able to use the data in the tag to rebuild the content yourself, so I think the plain `article` with the width/height we gathered is the way to go (which seems like what you were proposing in your last comment).